src/containers/SocketContainer.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component, createElement} from 'react';
+import { socketResponse } from '../actions/SocketAction'
+import createSocket, { Sockette } from "sockette-component";
+import { connect } from 'react-redux';
+
+const Socket = createSocket({
+  Component,
+  createElement
+});
+
+class SocketContainer extends Component {
+    // Socket internal state
+    state = {
+        speech: null,
+        endpoint: null,
+        socket: null
+    }
+
+    componentDidUpdate() {
+        if (this.props.text != "" && this.props.isRecording == false) {
+            this.sendMessage();
+        }
+    }
+
+    onOpen = ev => {
+        console.log("> Connected!", ev);
+    };
+
+    onMessage = ev => {
+        console.log("> Received:", ev.data);
+        this.props.socketResponse(ev.data)
+        //TODO: dispatch action with this ev.data
+        
+    };
+
+    onReconnect = ev => {
+        console.log("> Reconnecting...", ev);
+    };
+
+    sendMessage = _ => {
+        // WebSocket available in state!
+        const request = { "speech": this.props.text, "endpoint": "https://hap2a5df4m.execute-api.us-east-1.amazonaws.com/dev/ping", "state": { "directory": "home" } }
```

Comment:
We can think of moving some api endpoints to a service or at least a unified resource